sdk/mobile-center-distribute/src/main/java/com/microsoft/azure/mobile/distribute/Distribute.java

```diff
@@ -1130,6 +1175,57 @@ public void onClick(DialogInterface dialog, int which) {
     }
 
     /**
+     * Show update setupp failed dialog.
+     */
+    @UiThread
+    private synchronized void showUpdateSetupFailedDialog(final String errorMessage) {
+
+        /* Check if we need to replace dialog. */
+        if (!shouldRefreshDialog(mUpdateSetupFailedDialog)) {
+            return;
+        }
+        MobileCenterLog.debug(LOG_TAG, "Show update setup failed dialog.");
+
+        if (mForegroundActivity != null) {
+            AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(mForegroundActivity);
+            dialogBuilder.setCancelable(false);
+            dialogBuilder.setTitle(R.string.mobile_center_distribute_update_failed_dialog_title);
+            dialogBuilder.setMessage(errorMessage);
+            dialogBuilder.setPositiveButton(R.string.mobile_center_distribute_update_failed_dialog_ignore, new DialogInterface.OnClickListener() {
+
+                @Override
+                public void onClick(DialogInterface dialog, int which) {
+                    PreferencesStorage.putString(PREFERENCE_KEY_UPDATE_SETUP_FAILED_PACKAGE_HASH_KEY, DistributeUtils.computeReleaseHash(mPackageInfo));
```

Comment:
State can change between a dialog is shown and the click or cancel, please follow the same code pattern with state management as in every other dialog.